### PR TITLE
Create registration page and implement functionality

### LINKED ISSUE #49: Add ability to create an account from the signup page
New users should be able to create accounts. This task includes the following:
- create a signup activity (view)
- Add form fields to create a new user
- Use a radio button to decide tutor vs student signup
- Add a "submit" button that when clicked, checks for form validation, and if it passes, creates a new entry in the DB corresponding to the selected radio button (student or tutor). 
- This view should bring the student / tutor home page (they should automatically get logged into their account after signing up).

### PR DESCRIPTION
**Pull request by ericdesj**
The crux of this task was to create a new activity for students or tutors to register to the application and implement the business logic to do so (i.e. validation + DB operations).

**Validations**
- Form fields are empty
- Email format
- Password and confirmation match
- Select a role: student or tutor

**Database**
- Get all the schools from the `School` table
- If all validations pass, we create a new tutor or student record

**Intents**
When the registration is successful, the user is redirected to his/her profile page based on their role (student or tutor).

**Todo**
Created `todo` to verify if email is already in use. Opened issue #54. I assigned this task to myself and @gregpmillr, once it is complete I will update the validations.

### Registration
student_register

### Validation
errors

Closes #49

### CODE REVIEW

File: app/src/main/java/cs/dal/krush/RegistrationActivity.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+ * the Krush application. It performs basic form validation
+ * (i.e. required fields) and creates a new user in the DB
+ * upon succesfully filling out the registration form.
+ *
+ */
+public class RegistrationActivity extends AppCompatActivity {
+
+    private int profileSelected;
+    private int schoolID;
+    private boolean valdated;
```
Comment (Mons1Oerjan) [inline app/src/main/java/cs/dal/krush/RegistrationActivity.java]: refactor to "validated"

Comment (ericdesj) [inline app/src/main/java/cs/dal/krush/RegistrationActivity.java]: A typo indeed! 👍

Comment (ericdesj) [inline app/src/main/java/cs/dal/krush/RegistrationActivity.java]: Pushed up the change - updated the var name too.

**Comment by Mons1Oerjan** [comment]
Approved and merged!

**Pull request merged**